Sources/OpenAPIURLSession/URLSessionTransport.swift

```diff
@@ -90,7 +90,18 @@ public struct URLSessionTransport: ClientTransport {
     }
 
     private func invokeSession(_ urlRequest: URLRequest) async throws -> (Data, URLResponse) {
-        #if canImport(FoundationNetworking)
+    #if canImport(FoundationNetworking)
+        return try await performDataTask(with: urlRequest)
+    #else
+        if #available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *) {
+            return try await configuration.session.data(for: urlRequest)
+        } else {
+            return try await performDataTask(with: urlRequest)
+        }
+    #endif
+    }
+    
+    private func performDataTask(with urlRequest: URLRequest) async throws -> (Data, URLResponse) {
```

Comment:
IIRC, the async data method isn't (or at least wasn't) available on Linux. But agreed, on Darwin, we should try to use it if it's available in all the newly supported OS versions.

Reply:
> IIRC, the async data method isn't (or at least wasn't) available on Linux.
Right, ofc, Linux. Can confirm that `data(for:)` is not available on Linux. 

> But agreed, on Darwin, we should try to use it if it's available in all the newly supported OS versions.k
I was actually interested to understand why we need this complexity and why we can't just use the method that's available on all the platforms we want to support, i.e. `dataTask(with:completionHandler:)`?

Reply:
@simonjbeaumont just to be clear, are you saying to drop the new async method introduced in iOS 15 and just use the old method for everything?

Reply:
I got your point now. Xcode let me down on the versions. I jumped to the definition on foundation and it brought me to a similar method but with delegate [URLSession.data(for:delegate:)](https://developer.apple.com/documentation/foundation/urlsession/3767352-data), which is only available on macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, which led to the confusion.

Yes, this check can be done only for linux, I will remove it for Darwin platforms.

Reply:
> I got your point now. Xcode let me down on the versions. I jumped to the definition on foundation and it brought me to a similar method but with an optional delegate param [URLSession.data(for:delegate:)](https://developer.apple.com/documentation/foundation/urlsession/3767352-data), which is only available on macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, which led to the confusion.
> 
> Yes, this check can be done only for linux, I will remove it for Darwin platforms.

That makes sense—I also had to double-take on those methods for the same reason.

> @simonjbeaumont just to be clear, are you suggesting to drop the new async method introduced in macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0 and just use the old method for everything?

Yes, that was my suggestion, but let's check @czechboy0 has no objections because I wouldn't want you to have to go back-and-forth 🙂

Reply:
I guess I wanted to make it clear that we're trying to use the async method, it's just not yet available on Linux.

But I see your point, @simonjbeaumont. We can use the same implementation everywhere, just please add a comment, @denil-ct, saying that we're not using the async data method, because it's not available on Linux. Otherwise folks might be unsure why we're not using the async method available in Darwin SDKs.

Reply:
Sure 🙌 
Also, the current implementation has empty data initialization and force unwrap of response, shall I change those as well?

Reply:
Have made the changes, please have a look.